src/main/java/org/mockito/internal/invocation/InvocationMatcher.java

```diff
@@ -134,24 +133,12 @@ private void captureVarargsPart(Invocation invocation) {
             return;
         }
 
-        Object[] rawArguments = invocation.getRawArguments();
-        int indexOfVarArg = rawArguments.length - 1;
+        Object[] arguments = invocation.getArguments();
 
-        Object varArg = rawArguments[indexOfVarArg];
-        if (varArg == null) {
-            return;
-        }
-        int varArgSize = Array.getLength(varArg);
-
-        if (matchers.size() == indexOfVarArg) {
-            return;
-        }
-
-        for (int i = 0; i < varArgSize; i++) {
-            int indexOfVarargMatcher = (i + indexOfVarArg > matchers.size() -1 ? indexOfVarArg : i + indexOfVarArg);
-            ArgumentMatcher m = matchers.get(indexOfVarargMatcher);
+        for (int i = 0; i < arguments.length; i++) {
+            ArgumentMatcher m = matchers.get(i > matchers.size() - 1 ? matchers.size() - 1 : i);
             if (m instanceof CapturesArguments) {
-                ((CapturesArguments) m).captureFrom(Array.get(varArg, i));
+                ((CapturesArguments) m).captureFrom(arguments[i]);
             }
         }
```

Comment:
@ChristianSchwarz 
Hi Christian,
I tried to fix this. I think the problem is in `ArgumentProcessor` -> `expandVarArgs`. If I try to fix here, I brake the backward compatibility. Do you have any ideas how to fix this?


```java
 public static Object[] expandVarArgs(final boolean isVarArgs, final Object[] args) {
        if (!isVarArgs || new ArrayUtils().isEmpty(args) || args[args.length - 1] != null && !args[args.length - 1].getClass().isArray()) {
            return args == null ? new Object[0] : args;
        }

        int nonVarArgsCount = args.length - 1;
        Object[] varArgs;
        if (args[nonVarArgsCount] == null) {
            // in case someone deliberately passed null varArg array
            varArgs = new Object[] { null };
            nonVarArgsCount = 0;
        } else {
            varArgs = ArrayEquals.createObjectArray(args[nonVarArgsCount]);
        }
        final int varArgsCount = varArgs.length;
        Object[] newArgs = new Object[nonVarArgsCount + varArgsCount];
        System.arraycopy(args, 0, newArgs, 0, nonVarArgsCount);
        System.arraycopy(varArgs, 0, newArgs, nonVarArgsCount, varArgsCount);
        return newArgs;
    }
```